[5.1] Provide better exception in managers for non-existent connectors

Lagawrd opened the pull request:
Encountered this when I was setting up my database cache.  In config/cache.php under stores.database I set the driver as mysql since that's what I thought would make sense.  In doing so, it returned a 500 error with no message or clue.  I had to knowingly break other parts of the code ( like using an unknown cache store ) to get an actual error message telling me at which file to look at.  I was then able to diagnose and fix the problem.

The error, with no message, seems to happen when a valid cache store was selected and an invalid driver was set.

I thought this would be helpful to those that have and will encounter this or a similar error.

Commit:
[5.1] Provide better exception in managers for non-existent connectors

GrahamCampbell commented on src/Illuminate/Cache/CacheManager.php:
```diff
@@ -94,7 +94,13 @@ protected function resolve($name)
             return $this->callCustomCreator($config);
         } else {
-            return $this->{'create'.ucfirst($config['driver']).'Driver'}($config);
+            $create_driver_method = 'create'.ucfirst($config['driver']).'Driver';
```
you must use camelCase for variable names

GrahamCampbell replied:
also, please choose a much shorter variable name

GrahamCampbell commented on src/Illuminate/Cache/CacheManager.php:
```diff
@@ -94,7 +94,13 @@ protected function resolve($name)
+            if (method_exists($this, $create_driver_method)) {
+                return $this->{'create'.ucfirst($config['driver']).'Driver'}($config);
+            } else {
+                throw new InvalidArgumentException("Cache driver method [{$create_driver_method}] is not defined.");
```
That exception makes little sense IMO.

GrahamCampbell replied:
Better to say that the driver is unknown, or something.

Lagawrd commented:
Agreed about the camelCase and much shorter variable, however I do think the exception is useful, it would have saved me an hour had a message returned with the 500.

Lagawrd commented:
Fixed.

GrahamCampbell commented on src/Illuminate/Cache/CacheManager.php:
```diff
@@ -94,7 +94,13 @@ protected function resolve($name)
+            $driverMethod = 'create'.ucfirst($config['driver']).'Driver';
+
+            if (method_exists($this, $driverMethod)) {
+                return $this->{'create'.ucfirst($config['driver']).'Driver'}($config);
```
just use the $driverMethod variable here

GrahamCampbell commented:
Also, does this not affect the other manager classes too?

Lagawrd commented:
Good question.  From first glance, the same issue seems to exist in FilesystemManager::resolve().  I am not able to think of a DRY method for all manager classes, we would probably have to repeat a similar check there as well.

Lagawrd commented:
I will create another pull request for the FilesystemManager class when I do some testing if everything else here is good.

GrahamCampbell commented:
There are other manager classes too.

lucasmichot commented:
`@throws \InvalidArgumentException` is missing in `\Illuminate\Cache\CacheManager::resolve` docblock

GrahamCampbell commented:
Please make the fixes to all manager classes and squash to one commit.

Lagawrd commented:
Thanks @lucasmichot 

Here are the manager classes I found
``` CacheManager ``` : Fixed

```FilesystemManager``` : Fixed

```QueueManager``` : Utilizes connectors array in ```QueueManager::getConnector()``` to check driver support before dynamically calling the class method, otherwise throws the needed exception.  So no need to add the logic here.

```AuthManager``` : No need to add logic here either since it already does a similar check in ```AuthManager::createDriver()``` by calling the parent  `Manager::createDriver()`.  In fact, for the sake of consistency, this is where I got the new error message.

`BroadcastManager` : Fixed

`DatabaseManager` : `DatabaseManager::makeConnection()` Utilizes extensions array in a similar way `QueueManager::getConnector()` makes use of the connectors array.  It also has its own fallback method if a driver isn't supported in `ConnectionFactory::createConnection()`

`SessionManager` : I don't think anything needs to be done here.

`TransportManager` : I don't think anything needs to be done here.  Furthermore, like 
`SessionManager`, it extends Manager class which already has similar driver checking logic.
